packages/adapter-drizzle-all/src/mysql/index.ts

```diff
@@ -0,0 +1,293 @@
+ *  sessionToken: text("sessionToken").notNull().primaryKey(),
+ *  expires: integer("expires").notNull(),
+ * })
+ *
+ * export const verificationTokens = pgTable("verificationToken", {
+ *  identifier: text("identifier").notNull(),
+ *  token: text("token").notNull(),
+ *  expires: integer("expires").notNull()
+ * }, (vt) => ({
+ *   _: primaryKey(vt.identifier, vt.token)
+ * }))
+ *
+ * const pool = new Pool({
+ *   connectionString: "YOUR_CONNECTION_STRING"
+ * });
+ *
+ * export const db = drizzle(pool);
+ *
+ * migrate(db, { migrationsFolder: "./drizzle" })
+ *
+ * ```
+ *
+ **/
+export function MySqlAdapter(client: typeof db): Adapter {
+  return {
+    createUser: async (data) => {
+      await client
+        .insert(users)
+        .values({ ...data, id: "123" })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, '123'))
+        .then(res => res[0])
+
+    },
+
+    getUser: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data))
+        .then(res => res[0])
+        ?? null
+    },
+    getUserByEmail: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.email, data))
+        .then(res => res[0])
+        ?? null
+    },
+    createSession: async (data) => {
+      await client
+        .insert(sessions)
+        .values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(
+          eq(
+            sessions.sessionToken,
+            data.sessionToken
+          )
+        )
+        .then(res => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return client.select({
+        session: sessions,
+        user: users
+      })
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data))
+        .innerJoin(users, eq(users.id, sessions.userId))
+        .then(res => res[0])
+        ?? null
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client
+        .update(users)
+        .set(data)
+        .where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(
+          eq(
+            users.id,
+            data.id
+          )
+        )
+        .then(res => res[0])
+
+
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then(res => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
+        .then(res => res[0])
+    },
+    getUserByAccount: async (account) => {
+      return client.select({
+        id: users.id,
```

Comment:
Inconsistency: `getUser` and `getUserByEmail` return whole user record, where `getUserByAccount` returns only selected fields